src/Test/QuickCheck/Instances.hs

```diff
@@ -352,7 +352,10 @@ instance CoArbitrary OldTime.CalendarTime where
 -------------------------------------------------------------------------------
 
 instance Arbitrary Time.Day where
-    arbitrary = Time.ModifiedJulianDay <$> (2000 +) <$> arbitrary
+    arbitrary = sized $ \n -> do
+      let limit = (toInteger n)*365
```

Comment:
`n` can be big, say `10000`, is it ok here?

Reply:
There doesn't seem to be any inherent problem with such large values:

```haskell
λ ModifiedJulianDay $ -10000*365
-8135-07-08
λ ModifiedJulianDay $ 10000*365
11852-03-28
λ UTCTime (ModifiedJulianDay $ -10000*365) 0
-8135-07-08 00:00:00 UTC
```